UsageEnvironment/Log.cpp

```diff
@@ -132,17 +248,27 @@ void _Log::Fatal(const char* filePath, int line, const char * format, ...) {
 }
 
 void _Log::Error(const char* filePath, int line, const char * format, ...) {
-  if( logLevel >= LogLevelError  ) {
+  if( logLevel >= LogLevelError ) {
     va_list argptr; va_start(argptr, format);
     FILE* fileHandle = AcquireFile();
-    vfprintf(fileHandle, FormatLog(filePath, line, LogLevelError, format).c_str(), argptr);
+    char errorLog[MAX_LOG_LEN];
+    vsnprintf(errorLog, MAX_LOG_LEN, FormatLog(filePath, line, LogLevelError, format).c_str(), argptr);
+    fprintf(fileHandle, "%s", errorLog);
```

Comment:
Why not just use the string returned from `FormatLog`? (Although a moot point since we won't be doing string comparisons after this function has been revised.)